tests/e2e/framework/testInfo.go

```diff
@@ -111,6 +134,71 @@ func (t testInfo) Update(r int) error {
 	return t.createStatusFile(r)
 }
 
+func (t testInfo) FetchAndSaveClusterLogs() error {
```

Comment:
I reverted e2e.sh and extract to another shell script (getProjectID.sh) the section that gets project id with gcloud command. 
Here, I check if projectID is empty. If so, I call getProjectID. Users can always supersede this value with --project_id flag, but reading the project id is important because the user may have multiple clusters configured and the getProjectID script always identifies the right cluster where the tests are being run, so users do not have to constantly keep tract this value.